iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/AbstractThreeWindingsTransformerTest.java

```diff
@@ -1038,4 +1041,69 @@ private void createThreeWindingsTransformerWithLeg3(double r, double x, double g
             .add()
             .add();
     }
+
+    @Test
+    public void testAdderByCopy() {
+        // First limit
+        ThreeWindingsTransformerAdder transformerAdder = createThreeWindingsTransformerAdder();
+        ThreeWindingsTransformer transformer = transformerAdder.add();
+        ThreeWindingsTransformer.Leg leg1 = transformer.getLeg1();
+
+        CurrentLimitsAdder currentLimitsAdder1 = leg1.newCurrentLimits()
+                .setPermanentLimit(1000.)
+                .beginTemporaryLimit()
+                .setName("TL1")
+                .setAcceptableDuration(20 * 60)
+                .setValue(1200.0)
+                .endTemporaryLimit();
+        currentLimitsAdder1.add();
+
+        ActivePowerLimitsAdder activePowerLimitsAdder1 = leg1.newActivePowerLimits()
+                .setPermanentLimit(1000.)
+                .beginTemporaryLimit()
+                .setName("TL1")
+                .setAcceptableDuration(20 * 60)
+                .setValue(1200.0)
+                .endTemporaryLimit();
+        activePowerLimitsAdder1.add();
+
+        ApparentPowerLimitsAdder apparentPowerLimitsAdder1 = leg1.newApparentPowerLimits()
+                .setPermanentLimit(1000.)
+                .beginTemporaryLimit()
+                .setName("TL1")
+                .setAcceptableDuration(20 * 60)
+                .setValue(1200.0)
+                .endTemporaryLimit();
+        apparentPowerLimitsAdder1.add();
+
+        CurrentLimits currentLimits1 = leg1.getCurrentLimits().get();
+        ActivePowerLimits activePowerLimits1 = leg1.getActivePowerLimits().get();
+        ApparentPowerLimits apparentPowerLimits1 = leg1.getApparentPowerLimits().get();
+
+        CurrentLimitsAdder currentLimitsAdder2 = leg1.newCurrentLimits(currentLimits1);
```

Comment:
It would be better to add the new limits on leg2 or leg3.
In the current state, if the adders don't do anything, the test will still succeed.